Client/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```diff
@@ -62,353 +69,120 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
-                    }
-                    lua_pushelement(luaVM, pMarker);
-                    return 1;
-                }
-            }
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    CElementGroup* elementGroup = resource->GetElementGroup();
+    if (elementGroup)
+        elementGroup->Add(radarMarker);
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return radarMarker;
 }
 
-int CLuaBlipDefs::GetBlipIcon(lua_State* luaVM)
+std::variant<CClientRadarMarker*, bool> CLuaBlipDefs::CreateBlipAttachedTo(lua_State* const luaVM, CClientEntity* const entity, const std::optional<std::uint8_t> icon, std::optional<std::uint8_t> size, const std::optional<std::uint8_t> r, const std::optional<std::uint8_t> g, const std::optional<std::uint8_t> b, const std::optional<std::uint8_t> a, const std::optional<std::int16_t> ordering, const std::optional<std::uint16_t> visibleDistance)
 {
-    CClientRadarMarker* pMarker = NULL;
-    CScriptArgReader    argStream(luaVM);
-    argStream.ReadUserData(pMarker);
+    if (icon.has_value() && !CClientRadarMarkerManager::IsValidIcon(icon.value()))
+        throw std::invalid_argument("Invalid icon");
 
-    if (!argStream.HasErrors())
+    if (size.has_value() && size.value() > MAX_BLIP_SIZE)
     {
-        unsigned char ucIcon = static_cast<unsigned char>(pMarker->GetSprite());
-        lua_pushnumber(luaVM, ucIcon);
-        return 1;
+        m_pScriptDebugging->LogWarning(luaVM, SString("Blip size beyond %i is no longer supported (got %i). It will be clamped between 0 and %i.", MAX_BLIP_SIZE, size.value(), MAX_BLIP_SIZE));
+        size = MAX_BLIP_SIZE;
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
-
-int CLuaBlipDefs::GetBlipSize(lua_State* luaVM)
-{
-    CClientRadarMarker* pMarker = NULL;
-    CScriptArgReader    argStream(luaVM);
-    argStream.ReadUserData(pMarker);
 
-    if (!argStream.HasErrors())
+    CResource* resource = &lua_getownerresource(luaVM);
+    if (!resource)
     {
-        unsigned char ucSize = static_cast<unsigned char>(pMarker->GetScale());
-        lua_pushnumber(luaVM, ucSize);
-        return 1;
+        m_pScriptDebugging->LogError(luaVM, "Couldn't find the resource element");
+        return false;
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
 
-int CLuaBlipDefs::GetBlipColor(lua_State* luaVM)
-{
-    CClientRadarMarker* pMarker = NULL;
-    CScriptArgReader    argStream(luaVM);
-    argStream.ReadUserData(pMarker);
+    CClientRadarMarker* radarMarker = CStaticFunctionDefinitions::CreateBlipAttachedTo(*resource, *entity, icon.value_or(0), size.value_or(2), SColorRGBA(r.value_or(255), g.value_or(0), b.value_or(0), a.value_or(255)), ordering.value_or(0), visibleDistance.value_or(16383));
```

Comment:
`16383`?

Reply:
This is the default value from the createBlip page on the mta wiki

Reply:
Please document it (or format the code);
16383 is int16_t max value / 2